web/explorer/src/components/RuleMatchesTable.vue

```diff
@@ -0,0 +1,289 @@
+const onRightClick = (event, instance) => {
+    if (instance.node.data.source) {
+        selectedNode.value = instance.node;
+        // contrust capa-rules url
+        selectedNode.value.url = `https://github.com/mandiant/capa-rules/blob/master/${instance.node.data.namespace || "lib"}/${instance.node.data.name.toLowerCase().replace(/\s+/g, "-")}.yml`;
+        // construct VirusTotal deep link
+        const behaviourSignature = `behaviour_signature:"${instance.node.data.name}"`;
+        selectedNode.value.vturl = `https://www.virustotal.com/gui/search/${encodeURIComponent(behaviourSignature)}/files`;
+
+        menu.value.show(event);
+    }
+};
+
+/*
+ * Expand node on click
+ */
+const onNodeSelect = (node) => {
+    const nodeKey = node.key;
+    const nodeType = node.data.type;
+
+    // We only expand rule and match locations, if not return
+    if (nodeType !== "rule" && nodeType !== "match location") return;
+
+    // If the node is already expanded, collapse it
+    if (expandedKeys.value[nodeKey]) {
+        delete expandedKeys.value[nodeKey];
+        return;
+    }
+
+    if (nodeType === "rule") {
+        // For rule nodes, clear existing expanded keys and set the clicked rule as expanded
+        // and expand the first (child) match by default
+        expandedKeys.value = { [nodeKey]: true, [`${nodeKey}-0`]: true };
+    } else if (nodeType === "match location") {
+        // For match location nodes, we need to keep the parent expanded
+        // and toggle the clicked node while collapsing siblings
+        const [parentKey] = nodeKey.split("-");
+        expandedKeys.value = { [parentKey]: true, [`${nodeKey}`]: true };
+    } else {
+        return;
+    }
+};
+
+// All available columns
+const visibleColumns = ref([
+    { field: "address", header: "Address" },
+    { field: "namespace", header: "Namespace" },
+    { field: "tactic", header: "ATT&CK Tactic" },
+    { field: "mbc", header: "Malware Behaviour Catalogue" }
+]);
+
+// Filter out the treeData for showing/hiding lib rules
+const filteredTreeData = computed(() => {
+    if (props.showLibraryRules) {
+        return treeData.value; // Return all data when showLibraryRules is true
+    } else {
+        // Filter out library rules when showLibraryRules is false
+        const filterNode = (node) => {
+            if (node.data && node.data.lib) {
+                return false;
+            }
+            if (node.children) {
+                node.children = node.children.filter(filterNode);
+            }
+            return true;
+        };
+        return treeData.value.filter(filterNode);
+    }
+});
+
+const showSource = (source) => {
+    currentSource.value = source;
+    sourceDialogVisible.value = true;
+};
+
+onMounted(() => {
+    if (props.data && props.data.rules) {
+        treeData.value = parseRules(props.data.rules, props.data.meta.flavor, props.data.meta.analysis.layout);
+    } else {
+        console.error("Invalid data prop:", props.data);
```

Comment:
better show a toast here or is this only interesting for devs?
I'm not sure what's the best in JS to log errors - I'd assume this shouldn't show up in the final build?

Reply:
This is removed.